externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala

```diff
@@ -139,6 +146,58 @@ class ExecuteStatement(
     setState(OperationState.FINISHED)
   }
 
+  private def runSetOperation(setOperation: SetOperation): Unit = {
+    if (setOperation.getKey.isPresent) {
+      val key: String = setOperation.getKey.get.trim
+
+      if (setOperation.getValue.isPresent) {
+        val newValue: String = setOperation.getValue.get.trim
+        executor.setSessionProperty(sessionId, key, newValue)
+      }
+
+      val value = executor.getSessionConfigMap(sessionId).getOrDefault(key, "")
+      resultSet = ResultSet.builder
+        .resultKind(ResultKind.SUCCESS_WITH_CONTENT)
+        .columns(Column.physical("set", DataTypes.STRING()))
+        .data(Array(Row.of(PROPERTY_FORMAT.format(key, value))))
+        .build
+    } else {
+      // show all properties if set without key
+      val properties: util.Map[String, String] = executor.getSessionConfigMap(sessionId)
+
+      val entries = ArrayBuffer.empty[Row]
+      properties.forEach((key, value) => entries.append(Row.of(PROPERTY_FORMAT.format(key, value))))
+
+      if (entries.nonEmpty) {
+        val prettyEntries = entries.sortBy(_.getField(0).asInstanceOf[String])
+        resultSet = ResultSet.builder
+          .resultKind(ResultKind.SUCCESS_WITH_CONTENT)
+          .columns(Column.physical("set", DataTypes.STRING()))
+          .data(prettyEntries.toArray)
+          .build
+      } else {
+        resultSet = ResultSet.builder
+          .resultKind(ResultKind.SUCCESS_WITH_CONTENT)
+          .columns(Column.physical("set", DataTypes.STRING()))
+          .data(Array[Row]())
+          .build
+      }
+    }
+    setState(OperationState.FINISHED)
+  }
+
+  private def runResetOperation(resetOperation: ResetOperation): Unit = {
+    if (resetOperation.getKey.isPresent) {
+      // reset the given property
+      executor.resetSessionProperty(sessionId, resetOperation.getKey.get())
+    } else {
+      // reset all properties
+      executor.resetSessionProperties(sessionId)
+    }
+    resultSet = OperationUtil.successResultSet()
```

Comment:
Does it make more sense to return the original value to which we reset the key?

Reply:
@yaooqinn I think it's a valid consideration, but now hive returns nothing for `reset` while flink returns `OK` (which is the same with some queries like DDL), so I think it's okay to be aligned with them. WDYT?

Reply:
I am OK with this if flink takes `OK` as its side effect for all DDLs